Package/Image/ImageSpecifier.cs

```diff
@@ -36,16 +99,34 @@ public class ImageSpecifier
         public ImageIdentifier Resolve(CancellationToken cancellationToken)
         {
             List<IPackageRepository> repositories = Repositories.Distinct().Select(PackageRepositoryHelpers.DetermineRepositoryType).GroupBy(p => p.Url).Select(g => g.First()).ToList();
-            
-            DependencyResolver resolver = new DependencyResolver(Packages, repositories, cancellationToken);
-
 
-            if (resolver.DependencyIssues.Any())
-                throw new ImageResolveException(resolver.GetDotNotation(), $"OpenTAP packages could not be resolved", resolver.DependencyIssues);
+            var cache = new PackageDependencyCache(OS, Architecture, Repositories);
+            cache.LoadFromRepositories();
+            cache.AddPackages(InstalledPackages);
+            cache.AddPackages(AdditionalPackages);
+            
+            var resolver = new ImageResolver(cancellationToken);        
+            var image = resolver.ResolveImage(this, cache.Graph);
+            if (image.Success == false)
+            {
+                var unsatisfiedDependencies = InstalledPackages.Where(x => false == x.Dependencies.All(dep =>
+                    InstalledPackages.Any(x2 =>
+                        x2.Name == dep.Name && dep.Version.IsSatisfiedBy(x2.Version.AsExactSpecifier())))).ToArray();
+                if (unsatisfiedDependencies.Any())
+                {
+                    throw new ImageResolveException(image,
+                        string.Format("This is probably due to the current following package dependencies being broken: {0}",
+                        string.Join(", ", unsatisfiedDependencies.Select(x => x.Name))));
```

Comment:
I agree the error message here is not the best.

Your suggestion for a very specific error message e.g" X depends on Y so this cannot work" in a consistent manner is very hard to implement. Mostly because the reality is not that clear cut - there is always at least two ways to state a dependency problem. See for example this:
```
./tap image install  -t ../../deply "OpenTAP:9.16,TUI:0.1.0-beta.161```
Unable to resolve packages: TUI: 0.1.0-beta.161
```
In this case we are unable to resolve TUI 0.1.0-beta.161. This is because that version depends on OpenTAP 9.18, but we explicitly specify a 9.16 version.

But depending on the order, you could also say that it is `OpenTAP:9.16` that we could not resolve.

What we are printing now is the missing dependencies in the last failed attempt to resolve the problem, but there could be many other places to fix the issue (e.g Updating OpenTAP instead of downgrading TUI).


Anyway, I have changed it so that it says "Unable to resolve packages" is that better?